bin/improver-nbhood-vicinity

```diff
@@ -0,0 +1,129 @@
+                    'within a vicinity via '
+                    'the ProbabilityOfOccurence plugin '
+                    'to a file with one cube.')
+    parser.add_argument('--neighbourhood_shape', metavar='NEIGHBOURHOOD_SHAPE',
+                        choices=["square"], default="square",
+                        help='The shape of the neighbourhood to apply in '
+                             'neighbourhood processing. Only a "square" '
+                             'neighbourhood shape is applicable for '
+                             'calculating the probability within a '
+                             'vicinity.')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--radius', metavar='RADIUS', type=float,
+                       help='The radius (in m) for neighbourhood processing.')
+    group.add_argument('--radii-by-lead-time',
+                       metavar=('RADII_BY_LEAD_TIME', 'LEAD_TIME_IN_HOURS'),
+                       nargs=2,
+                       help='The radii for neighbourhood processing '
+                       'and the associated lead times at which the radii are '
+                       'valid. The radii are in metres whilst the lead time '
+                       'has units of hours. The radii and lead times are '
+                       'expected as individual comma-separated lists with '
+                       'the list of radii given first followed by a list of '
+                       'lead times to indicate at what lead time each radii '
+                       'should be used. For example: 10000,12000,14000 1,2,3 '
+                       'where a lead time of 1 hour uses a radius of 10000m, '
+                       'a lead time of 2 hours uses a radius of 12000m, etc.')
+    parser.add_argument('--ens_factor', metavar='ENS_FACTOR', type=float,
+                        default=1.0,
+                        help='The factor with which to adjust the '
+                        'neighbourhood size for more than one '
+                        'ensemble member. If ens_factor = 1.0 this '
+                        'essentially conserves ensemble members if '
+                        'every grid square is considered to be the '
+                        'equivalent of an ensemble member.'
+                        'Optional, defaults to 1.0.')
+    parser.add_argument('--weighted_mode', action='store_true', default=False,
+                        help='For neighbourhood processing using a circular '
+                             'kernel, setting the weighted_mode indicates the '
+                             'weighting decreases with radius. '
+                             'If weighted_mode is not set, a constant '
+                             'weighting is assumed. Currently this '
+                             'keyword does nothing as only a square '
+                             'kernel is applicable.')
+    parser.add_argument('vicinity_distance', metavar='VICINITY_DISTANCE',
```

Comment:
I believe that as a general rule that we'd put the optional arguments after the essential ones.

Reply:
I was following improver-nbhood which seems to have it the other way around.